src/main/kotlin/com/wgslfuzz/core/Resolver.kt

```diff
@@ -510,71 +557,87 @@ private fun nodeIntroducesNewScope(
             parentNode !is ContinuingStatement
     )
 
+private fun <T : Type> T.asStoreType(): Type =
```

Comment:
Do we derive any benefit from having the signature:

```
<T : Type> T.asStoreType(): Type
```

rather than simply

```
Type.asStoreType(): Type
```

?

I think the latter would be applicable anywhere the former is, because `T` is not used as an argument or return type.